numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -204,8 +204,31 @@ nonparametric_default_descr(PyArray_DTypeMeta *cls)
 }
 
 
+/* Ensure a copy of the singleton (just in case we do adapt it somewhere) */
 static PyArray_Descr *
-flexible_default_descr(PyArray_DTypeMeta *cls)
+datetime_and_timedelta_default_descr(PyArray_DTypeMeta *cls)
+{
+    return PyArray_DescrNew(cls->singleton);
+}
+
+
+static PyArray_Descr *
+void_default_descr(PyArray_DTypeMeta *cls)
+{
+    PyArray_Descr *res = PyArray_DescrNew(cls->singleton);
+    if (res == NULL) {
+        return NULL;
+    }
+    /*
+     * The legacy behaviour for `np.array([], dtype="V")` is to use "V8".
+     * This is because `[]` uses `float64` as dtype, and then that is used
+     * for the size of the requested void.
+     */
+    res->elsize = 8;
```

Comment:
Needs to return something :)

Reply:
Probably ;), strange I was sure I ran the array-coercion tests on this. Thanks.

Reply:
Fixed, but the tests seem to pass even if commenting it out, so have to check, must be something wrong with the tests.

Reply:
OK, looks all fine. Seems through some stack-magic, the return value ended up being `res` even though the return was missing...

Reply:
I think I recall that the last value gets returned by default in these cases. But it seems that isn't right ;)

> If the } that terminates a function is reached, and the value of the function call is used by the caller, the behavior is undefined.